lib/cell.rb

```diff
@@ -0,0 +1,49 @@
+class Cell
+
+  attr_reader :coordinate, :empty, :ship, :fired_upon
+
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @empty = true
+    @ship = nil
+    @fired_upon = false
```

Comment:
I tried not to make these instance variables but I couldn't figure out a workaround so here we are. I set the default for @ship to nil because if there is no ship, according to the instructions, it should return nil.

Reply:
I see no issue with making them instance variables. If it works then im good with it!